dind: add helper scripts to speed-up development with bind mounts

lekkas opened the pull request:
Workflow examples:

##### 1. Simply deploy and run supervisor-dind (no bind mounting)

```bash
~# ./tools/dev/dindctl deployrun
```

##### 2. Deploy and run supervisor-dind with `./src` bind-mounted into the running supervisor container:

```bash
~# ./tools/dev/dindctl deployrun --mount-src
```

##### 3. Deploy and run supervisor-dind with `./src` *and* `./node_modules` bind-mounted into the running supervisor container:

```bash
~# ./tools/dev/dindctl deployrun --mount-src --mount-nm
```

##### 4. Quickly reflect changes from local, bind-mounted `./src` to the running application container:

```bash
~# time ./tools/dev/dindctl refresh                                           
 * Compiling CoffeeScript..
 * Restarting supervisor container..
resin_supervisor
./tools/dev/dindctl refresh  1.48s user 0.18s system 14% cpu 11.475 total
```

Connects to #184

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/resin-io/resin-supervisor/192)
<!-- Reviewable:end -->

pcarranzav commented on tools/dev/dev_s.sh:
```diff
@@ -0,0 +1,101 @@
+		ARCH="$ARCH" \
+		PASSWORDLESS_DROPBEAR="$PASSWORDLESS_DROPBEAR" \
+		SUPERVISOR_EXTRA_MOUNTS="$SUPERVISOR_EXTRA_MOUNTS" \
+		SUPERVISOR_IMAGE="#{DEPLOY_REGISTRY}resin/${ARCH}-supervisor:master" \
```
Can we allow users to specify `SUPERVISOR_IMAGE` directly? I usually do this to run the images built by Jenkins for a specific PR.

pcarranzav commented:
Might be good to rename `dev_s.sh` into something more informative?

lekkas commented:
- Renamed helper script to `./tools/dev/dindctl`
- Added `logs [-f]` action to print out/follow supervisor logs
- Moved the code for the `refresh` action into the Makefile, since I think it makes sense to have all container-handling code in one place - `dindctl` is supposed to be a thin Makefile wrapper with sensible default options
- Also removed the `DEPLOY_REGISTRY` configuration variable and replaced it with `SUPERVISOR_IMAGE`instead - after all `SUPERVISOR_IMAGE` already includes the registry.

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[tools/dev/dindctl, line 73 \[r1\]](https://reviewable.io:443/reviews/resin-io/resin-supervisor/192#-KLx9wLDdcBsSAMJG98m-r1-73:-KLzYxP-qCiaI7ggONor:676635175) ([raw file](https://github.com/resin-io/resin-supervisor/blob/05bc529034587a3f711f86514e14f9f924302b1b/tools/dev/dindctl#L73)):*
<details><summary><i>Previously, pcarranzav (Pablo Carranza Vélez) wrote…</i></summary>
> Can we allow users to specify `SUPERVISOR_IMAGE` directly? I usually do this to run the images built by Jenkins for a specific PR.
</details>
Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/resin-io/resin-supervisor/192#-:-KLzXjwzp6E6k5OrsLEr:-940750850)*
<!-- Sent from Reviewable.io -->

Page- commented:
Reviewed 1 of 4 files at r1, 2 of 3 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[tools/dev/dindctl, line 5 \[r4\]](https://reviewable.io:443/reviews/resin-io/resin-supervisor/192#-KM0NE6pGIjcKkaTXoha:-KM0NE6pGIjcKkaTXohb:-1662174776) ([raw file](https://github.com/resin-io/resin-supervisor/blob/5cce7bec744b494b5cf58d60bdda0a63ac7730fd/tools/dev/dindctl#L5)):*
> ```
> set -o errexit
> set -o pipefail
> # set -o xtrace
> ```

It's probably best to just remove this rather than comment, as a comment it adds noise (and adding `set -x` when debugging is pretty standard)
Alternatively you could use something like
```
if [ -n "$DEBUG" ]; then
  set -o xtrace
fi
```
in order to enable it behind a flag

---

*[tools/dev/dindctl, line 72 \[r4\]](https://reviewable.io:443/reviews/resin-io/resin-supervisor/192#-KM0NkaGPQJvRQ573Ulu:-KM0NkaH4ZNWoiPHh9j0:-1753960105) ([raw file](https://github.com/resin-io/resin-supervisor/blob/5cce7bec744b494b5cf58d60bdda0a63ac7730fd/tools/dev/dindctl#L72)):*
> ```
> 				;;
> 			*)
> 				;;
> ```

Might be worth printing a warning about an unknown arg, in case someone hits a typo by accident

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/resin-io/resin-supervisor/192)*
<!-- Sent from Reviewable.io -->

lekkas commented:
Review status: 2 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[tools/dev/dindctl, line 5 \[r4\]](https://reviewable.io:443/reviews/resin-io/resin-supervisor/192#-KM0NE6pGIjcKkaTXoha:-KM14Uj-y3oEv1taN5aT:1241772465) ([raw file](https://github.com/resin-io/resin-supervisor/blob/5cce7bec744b494b5cf58d60bdda0a63ac7730fd/tools/dev/dindctl#L5)):*
<details><summary><i>Previously, Page- wrote…</i></summary>
> It's probably best to just remove this rather than comment, as a comment it adds noise (and adding `set -x` when debugging is pretty standard)
> Alternatively you could use something like
> ```
> if [ -n "$DEBUG" ]; then
>   set -o xtrace
> fi
> ```
> in order to enable it behind a flag
</details>
Good point, I removed it altogether.

---

*[tools/dev/dindctl, line 72 \[r4\]](https://reviewable.io:443/reviews/resin-io/resin-supervisor/192#-KM0NkaGPQJvRQ573Ulu:-KM14g_13VBvTDYn7g7d:-283038703) ([raw file](https://github.com/resin-io/resin-supervisor/blob/5cce7bec744b494b5cf58d60bdda0a63ac7730fd/tools/dev/dindctl#L72)):*
<details><summary><i>Previously, Page- wrote…</i></summary>
> Might be worth printing a warning about an unknown arg, in case someone hits a typo by accident
</details>
Done - added a warning

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/resin-io/resin-supervisor/192)*
<!-- Sent from Reviewable.io -->

Page- commented:
Reviewed 1 of 3 files at r3, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/resin-io/resin-supervisor/192)*
<!-- Sent from Reviewable.io -->

pcarranzav commented:
Review status: 2 of 3 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[Makefile, line 89 \[r6\]](https://reviewable.io:443/reviews/resin-io/resin-supervisor/192#-KMAKG84Oh3deav0VGE6:-KMAKG84Oh3deav0VGE7:1202493988) ([raw file](https://github.com/resin-io/resin-supervisor/blob/04315ee70e207a6bbf02952869eca9f287de2e82/Makefile#L89)):*
> ```Makefile
> run-supervisor: supervisor-dind stop-supervisor
> 	cd tools/dind \
> 	&& echo "SUPERVISOR_IMAGE=$(SUPERVISOR_IMAGE)\nPRELOADED_IMAGE=$(PRELOADED_IMAGE)\nSUPERVISOR_EXTRA_MOUNTS=$(SUPERVISOR_EXTRA_MOUNTS)" > config/localenv \
> ```

While we're at it, could you change this to use 3 separate echo commands? I've seen at least one case where either the make or echo versions cause that the `\n` is not interpreted as it should.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/resin-io/resin-supervisor/192)*
<!-- Sent from Reviewable.io -->

lekkas commented:
Review status: 2 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[Makefile, line 89 \[r6\]](https://reviewable.io:443/reviews/resin-io/resin-supervisor/192#-KMAKG84Oh3deav0VGE6:-KMAUEPiGWoEDi3yOdpE:676635175) ([raw file](https://github.com/resin-io/resin-supervisor/blob/04315ee70e207a6bbf02952869eca9f287de2e82/Makefile#L89)):*
<details><summary><i>Previously, pcarranzav (Pablo Carranza Vélez) wrote…</i></summary>
> While we're at it, could you change this to use 3 separate echo commands? I've seen at least one case where either the make or echo versions cause that the `\n` is not interpreted as it should.
</details>
Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/resin-io/resin-supervisor/192)*
<!-- Sent from Reviewable.io -->

pcarranzav commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 4 files at r1, 1 of 3 files at r3, 1 of 1 files at r5, 1 of 1 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/resin-io/resin-supervisor/192#-:-KMAUk333PDzZlWOzoVO:1417088721)*
<!-- Sent from Reviewable.io -->
